api/v1beta1/types.go

```diff
@@ -123,6 +161,15 @@ type LoadBalancerSpec struct {
 	// +kubebuilder:validation:Pattern=`(^[1-9][0-9]{0,31}$)|(^[a-z][a-z0-9-]{4,28}[a-z0-9]$)`
 	// +optional
 	APIServerInstanceGroupTagOverride *string `json:"apiServerInstanceGroupTagOverride,omitempty"`
+
+	// LoadBalancerType defines the type of Load Balancer that should be created.
+	// If not set, a Global External Proxy Load Balancer will be created by default.
+	// +optional
+	LoadBalancerType *LoadBalancerType `json:"loadBalancerType,omitempty"`
```

Comment:
Can the `LoadBalancerType` be defined as `[]LoadBalancerType`. This array can hold `InternalLoadBalancerOnly` and/or `ExternalLoadBalancerOnly`. This also eliminates the need for `DualLoadBalancer` since the array with both Internal and External LoadBalancers would be equivalent to Dual.

Reply:
I had thought about but felt that type of API presents too much ambiguity and potential for error.  For example with that type of array you could have the following potential combinations, all of which are in error:
- `empty` (no LBs defined)
-` INTERNAL/INTERNAL`
- `EXTERNAL/EXTERNAL`

Explicitly defining the Load Balancer configuration options seems more useful.